### DISCUSSION TITLE
Unable to load panel network module 'dbus' has no attribute 'mainloop'

### DISCUSSION THREAD

**Discussion by erdguez** [KlipperScreen/KlipperScreen#1066] [Q&A] [answered]
Anyone encountered this error opening the network panel??

`2023-07-31 22:29:51,609 [screen.py:_load_panel()] - Loading panel: network
2023-07-31 22:29:51,629 [network.py:__init__()] - Found wireless interfaces: ['wlp2s0f0']
2023-07-31 22:29:51,629 [network.py:__init__()] - Using NetworkManager
2023-07-31 22:29:51,694 [screen.py:show_error_modal()] - Showing error modal: Unable to load panel network module 'dbus' has no attribute 'mainloop'
2023-07-31 22:29:51,720 [KlippyGtk.py:Dialog()] - Showing dialog <Gtk.Dialog object at 0x7ff942f9ea80 (GtkDialog at 0x55b931d988b0)>`

any idea how to solve it??

EXTRA INFO:

`dpkg -s network-manager
Package: network-manager
Status: install ok installed
Priority: optional
Section: net
Installed-Size: 7232
Maintainer: Ubuntu Developers <ubuntu-devel-discuss@lists.ubuntu.com>
Architecture: amd64
Version: 1.36.6-0ubuntu2`

**Comment by alfrix** [comment, marked as answer]
it does work correctly for me, could you try this in ks_includes/wifi_nm.py (restart ks):
```diff
 from contextlib import suppress
 from ks_includes.wifi import WifiChannels
 from ks_includes import NetworkManager
+from dbus.mainloop.glib import DBusGMainLoop
 
 
 class WifiManager:
@@ -18,7 +19,7 @@ class WifiManager:
 
     def __init__(self, interface_name, *args, **kwargs):
         super().__init__(*args, **kwargs)
-        dbus.mainloop.glib.DBusGMainLoop(set_as_default=True)
+        DBusGMainLoop(set_as_default=True)
         self._callbacks = {
             "connected": [],
             "connecting_status": [],
```

**Reply by erdguez** [reply]
yes, this fix the crash here, thanks

**Reply by erdguez** [reply]
KS is not crashing but the panel is missing wifi info WPA etc...

im not running on a PI but a laptop asus x200ca 

any idea?
  
![IMG_7980](https://github.com/KlipperScreen/KlipperScreen/assets/755677/aca0938a-8026-4c21-a8b1-a8912a8c78b4)

**Comment by alfrix** [comment]
check the logfile, usually it's a permission issue:
https://klipperscreen.readthedocs.io/en/latest/Troubleshooting/Network/

**Comment by erdguez** [comment]
Never mind, it was work, it just need patience! 
 takes like 90sec to populate the available wifi networks...
all good
